packages/lodestar/src/sync/regular/naive/naive.ts

```diff
@@ -180,24 +191,28 @@ export class NaiveRegularSync extends (EventEmitter as { new(): RegularSyncEvent
   private waitForBestPeer = async (signal: AbortSignal): Promise<void> => {
     // statusSyncTimer is per slot
     const waitingTime = this.config.params.SECONDS_PER_SLOT * 1000;
-    const state = await this.chain.getHeadState();
     let isAborted = false;
     signal.addEventListener("abort", () => {
       this.logger.verbose("RegularSync: Abort waitForBestPeer");
       isAborted = true;
     });
+    const headSlot = this.chain.forkChoice.headBlockSlot();
     while (!this.bestPeer && !isAborted) {
       // wait first to make sure we have latest status
       await sleep(waitingTime);
-      const previousSlot = getCurrentSlot(this.config, state.genesisTime) - 1;
       const peers = this.network.getPeers();
-      const {slot: maxHeadSlot} = getBestHead(peers, this.reps);
-      if (maxHeadSlot >= previousSlot) {
-        this.bestPeer = getBestPeer(this.config, peers, this.reps);
-        this.logger.verbose(`Regular Sync: Found best peer ${this.bestPeer.toB58String()}, headSlot=${maxHeadSlot}`);
+      this.bestPeer = getBestPeer(this.config, peers, this.reps);
```

Comment:
We cannot sync 20k slots from one peer, we need to change logic to find best head and get all peers on same head.

Reply:
this is naive regular sync anyway, I prefer to fix to let it work first. Does v2 implement this?

even through we can get multiple peers with same head, it's not a guarantee we can get blocks from same chain since they can have different heads later, that's the concern we have in #1234 and we decided to get blocks from single best peer for this "naive" approach.

Reply:
@wemeetagain any ideas?

Reply:
I don't think syncing from a single peer would be so bad, if we could recover from that peer disconnecting us / timing out.
It seems we currently only do peer selection once, upfront.
I think the bigger thing we should add is the notion of regular sync failure, which should then recover from.
Strawman proposal: Regular sync knows when sync peers are no longer viable (eg: using heuristic N timeouts or disconnection), recalculates target + sync peers.

Reply:
with the latest update, whenever we get next range we call `getSyncPeers()` which then call `checkBestPeer()`. We'll find new best peer if
+ peer is not connected to node anymore
+ or peer head is not better than us